### PR TITLE
Fix.permission update fsp

### PR DESCRIPTION
**Pull request by jannisvisser**
[AB#31140](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/31140)

## Describe your changes

- apply permission check for update fsp on patch registration
- api-test on bulk update fsp

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/src/registration/registrations.controller.ts [hunk 1]
```diff
@@ -376,32 +377,42 @@ export class RegistrationsController {
...
-        if (
-          !FinancialAttributes.includes(
-            attributeKey as keyof RegistrationEntity,
-          )
-        ) {
+      } else if (
+        attributeKey ===
+        GenericRegistrationAttributes.programFinancialServiceProviderConfigurationName
+      ) {
+        if (!hasUpdateFspPermission) {
```
Comment (RubenGeo) [inline services/121-service/src/registration/registrations.controller.ts:410]: - Should we also have a test that covers this?

File: services/121-service/src/registration/registrations.controller.ts [hunk 2]
```diff
@@ -376,32 +377,42 @@ export class RegistrationsController {
         programId,
         PermissionEnum.RegistrationAttributeFinancialUPDATE,
       );
+    const hasUpdateFspPermission =
+      await this.registrationsPaginateService.userHasPermissionForProgram(
+        userId,
+        programId,
+        PermissionEnum.RegistrationFspUPDATE,
```
Comment (RubenGeo) [inline services/121-service/src/registration/registrations.controller.ts:384]: - Should we change the name of this permission to something with Config in it?

File: services/121-service/test/helpers/assert.helper.ts [hunk 3]
```diff
@@ -22,7 +22,10 @@ export function processMessagePlaceholders(
 }
 
 export function assertRegistrationBulkUpdate(
-  patchData: Record<string, string | undefined | boolean | number | null>,
+  patchData: Record<
+    string,
+    string | undefined | boolean | number | null | object
```
Comment (RubenGeo) [inline services/121-service/test/helpers/assert.helper.ts:27]: Should object here be LocalizedString?

File: services/121-service/test/registrations/bulk-update-registration.test.ts [hunk 4]
```diff
@@ -108,6 +108,84 @@ describe('Update attribute of multiple PAs via Bulk update', () => {
...
+    const pa2BeforePatch = searchByReferenceIdBeforePatchPa2.body.data[0];
+
+    // Act
+    const bulkUpdateResult = await bulkUpdateRegistrationsCSV(
+      programIdOcw,
+      './test-registration-data/test-registrations-patch-OCW-chosen-FSP.csv',
+      accessToken,
+      'test-reason',
+    );
+    console.log('bulkUpdateResult: ', bulkUpdateResult.body);
```
Comment (RubenGeo) [inline services/121-service/test/registrations/bulk-update-registration.test.ts:154]: Classic leftover console.log

**Review by RubenGeo — CHANGES_REQUESTED** [review]

File: services/121-service/src/registration/registrations.controller.ts (hunk 1 above)
Comment (jannisvisser) [reply services/121-service/src/registration/registrations.controller.ts:410]: added

**Review by RubenGeo — APPROVED** [review]

**Pull request merged**